pkg/provider/provider_helpers.go

```diff
@@ -0,0 +1,137 @@
+package provider
+
+import (
+	"crypto/rsa"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mitchellh/go-homedir"
+	"github.com/youmark/pkcs8"
+	"golang.org/x/crypto/ssh"
+)
+
+func mergeSchemas(schemaCollections ...map[string]*schema.Resource) map[string]*schema.Resource {
+	out := map[string]*schema.Resource{}
+	for _, schemaCollection := range schemaCollections {
+		for name, s := range schemaCollection {
+			out[name] = s
```

Comment:
Shouldn't we check for any unexpected overwriting here ?

Reply:
we have had this code in the provider for years now, and it has not caused a problem, so i don't think we need to change it. i was contemplating removing this entirely, as its only purpose is to segregating grant resource from other resources.  Once we are using the provider v2, then this can be removed.